app/src/main/java/com/example/ithaca_transit_android_v2/NetworkUtils.kt

```diff
@@ -1,12 +1,136 @@
 package com.example.ithaca_transit_android_v2
 
-import com.appdev.futurenovajava.Endpoint;
-import com.appdev.futurenovajava.FutureNovaRequest;
+import android.util.JsonReader
+import android.util.JsonToken
+import android.util.Log
+import com.example.ithaca_transit_android_v2.models.Location
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import okhttp3.*
+import java.io.IOException
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.MediaType.Companion.toMediaType
+import org.json.JSONObject
+import com.squareup.moshi.Types.newParameterizedType
+import android.R
+import com.example.ithaca_transit_android_v2.Models.DataLocation
+import com.example.ithaca_transit_android_v2.Models.LocationType
+import com.example.ithaca_transit_android_v2.models.Coordinate
+import com.squareup.moshi.*
 
 /*
  * NetworkUtils include all the networking calls needed
  */
 class NetworkUtils {
 
+    fun getSearchedLocations(
+        client: OkHttpClient,
+        url: String,
+        query: String
+    ): List<Location> {
+
+        val json = JSONObject()
+        json.put("query", query)
+        val mediaType = ("application/json; charset=utf-8").toMediaType()
+        var locListReturn: List<Location> = emptyList()
+
+        val requestBody = json.toString().toRequestBody(mediaType)
+
+        val request: Request = Request.Builder()
+            .url(url + "search")
+            .post(requestBody)
+            .build()
+
+        client.newCall(request).enqueue(object : Callback {
+            override fun onFailure(call: Call, e: IOException) {
+                Log.e("networking", "failed search location POST")
+            }
+
+            override fun onResponse(call: Call, response: Response) {
+                val body = response.body?.string()
+                println(body)
+
+                val type = newParameterizedType(List::class.java, Location::class.java)
+                val moshi = Moshi.Builder()
+                    .add(LocationAdapter())
+                    .add(KotlinJsonAdapterFactory())
+                    .build()
+
+                val adapter: JsonAdapter<List<Location>> = moshi.adapter(type)
+
+                val locList: List<Location>? = (adapter.fromJson(body))
+                // Test to see if we are successfully wrapping json into objects
+                Log.d("Return Value", locList.toString())
+
+                if (locList != null) {
+                    locListReturn = locList
+                }
+
+            }
+        })
+
+        return locListReturn
+    }
+}
+
+@JsonClass(generateAdapter = true)
```

Comment:
This entire class could be removed if we stuck with the same scheme as backend. I'll explain more in the final comment on this PR